### PR TITLE
EDGE-421 Provide detailed logging on error inside Connect SDK

### PR DESCRIPTION
**Pull request by ionyshch-prismsl**

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -214,56 +214,94 @@ bool hasIntMember(const rapidjson::Value& value, const char* name)
...
+        LOG(ERROR) << fname << ": " << rv;
+
+    return rv;
 }
 
-// TODO change prefix parse for something more suitable
-Status parseAccount(const rapidjson::Value& itemJson, Account& account)
+Status parseAccountJson(const rapidjson::Value& itemJson, Account& account)
 {
+    const char* fname = __FUNCTION__;
```
Comment (dbus-prsm) [inline src/client.cpp:300]: [__FUNCTION__ is another name for __func__. Older versions of GCC recognize only this name. However, it is not standardized. For maximum portability, we recommend you use __func__, but provide a fallback definition with the preprocessor](https://gcc.gnu.org/onlinedocs/gcc-4.8.1/gcc/Function-Names.html)

File: src/client.cpp [hunk 2]
```diff
@@ -402,435 +506,610 @@ Status parseInstrument(const rapidjson::Value& itemJson, Instrument& instrument)
...
+        if (cs->getResponseCode() != 201)
+        {
+            LOG(ERROR) << fname << ": POST " << url << " failed. "
+                       << cs->getResponseCode() << ", error message: "
+                       << cs->getErrorMessage();
+            rv = makeError(cs->getResponseCode(), Status::FACILITY_HTTP);
+            break;
+        }
+    } while (false);
+
```
Comment (dbus-prsm) [inline src/client.cpp:1111]: Shouldn't we have `rv` posted into the log here?

**Review by dbus-prsm — APPROVED** [review]

**Comment by kdeming** [comment]
You may want to check your notification settings on this repo. I still get
messages now and then.

File: src/client.cpp (hunk 1 above)
Comment (Yurmusp) [reply src/client.cpp:300]: It is said that now `__func__`  is the standard way, not `__FUNCTION__`.
At least it is defined in C99 and C++11. While `__FUNCTION__` is not standardized at all.
It is just mostly supported everywhere de-facto.

**Review by Yurmusp — COMMENTED** [review]
I do not see any major issues. `__func__` is the only thing to note. Also, looks like many changes.
Otherwise looks OK to me.

**Pull request merged**